### PR TITLE
[frontend/backend] Fix changing entity name to same as alias

### LINKED ISSUE #4798: Cannot modify the name of an entity if already present in the alias of the same entity
## Description

Cannot modify the name of an entity if already present in the alias of the same entity.

## Reproducible Steps

1. Create an intrusion set "Intrusion Set A"
2. Add the alias "Intrusion Set B" and "Intrusion Set A"
3. Try to modify the name with "Intrusion Set B"

### PR DESCRIPTION
**Pull request by Megafredo**
### Proposed changes

* Fix changing entity name to same as alias 
* Fix creating alias name to same as entity

### Related issues

* https://github.com/OpenCTI-Platform/opencti/issues/4798

### Checklist

- [x] I consider the submitted work as finished
- [x] I tested the code for its functionality
- [ ] I wrote test cases for the relevant uses case
- [ ] I added/update the relevant documentation (either on github or on notion)
- [ ] Where necessary I refactored code to improve the overall quality

### Further comments

Behavior before fix :
OpenCTI_alias_before_fix

Behavior after fix : 
OpenCTI_alias_after_fix

### CODE REVIEW

**Comment by labo-flg** [comment]
Would it make more sense for the user (in terms of UX) to have a validation in the form instead of "not doing the requested changes" being his back?

I mean, you edit the name, put one of its alias, the form is red with proper error like "Cannot rename to an alias".

IMO disallowing bad behavior with explanation is better UX (educating user).

**Comment by Megafredo** [comment]
Yes indeed, having a visual way to communicate why adding the alias does not work can be a good approach (in terms of UX).

File: opencti-platform/opencti-front/src/private/components/common/stix_domain_objects/StixDomainObjectHeader.jsx [hunk 1]
```diff
@@ -240,6 +240,15 @@ const StixDomainObjectHeader = (props) => {
...
   const onSubmitCreateAlias = (values, { resetForm, setSubmitting }) => {
     const currentAliases = getCurrentAliases();
+    const normalizeNameEntity = (stixDomainObject.name).toLowerCase().trim();
+    const normalizeNewAlias = newAlias.toLowerCase().trim();
+    if (normalizeNameEntity === normalizeNewAlias) {
+      setOpenAlias(false);
+      setOpenCommitCreate(false);
+      setNewAlias('');
+      resetForm();
+      return;
```
Comment (SouadHadjiat) [inline opencti-platform/opencti-front/src/private/components/common/stix_domain_objects/StixDomainObjectHeader.jsx:250]: I suggest we display an error message here to notify the user that the alias can't be named the same as the entity name.

**Review by labo-flg — APPROVED** [review]
Considering we leave the UX improvements for later, it looks ok to me !

File: opencti-platform/opencti-graphql/src/database/middleware.js [hunk 2]
```diff
@@ -1801,8 +1802,19 @@ const updateAttributeRaw = async (context, user, instance, inputs, opts = {}) =>
 export const updateAttributeMetaResolved = async (context, user, initial, inputs, opts = {}) => {
   const { locks = [], impactStandardId = true } = opts;
   const updates = Array.isArray(inputs) ? inputs : [inputs];
+
+  const filteredUpdates = updates.map((item) => {
+    if (item.key === ATTRIBUTE_ALIASES || item.key === ATTRIBUTE_ALIASES_OPENCTI) {
+      const filteredValues = item.value.filter(e => normalizeName(e) !== normalizeName(initial.name));
```
Comment (SouadHadjiat) [inline opencti-platform/opencti-graphql/src/database/middleware.js:1808]: are we sure item.value is defined and is a list ?

**Comment by richard-julien** [comment]
Just thinking with Souad about a little missing piece at creation.
Alias should also be purged of the name when creating the entity.
Dont hesitate to join me to have more information @Megafredo

**Pull request merged**